Buildings/Fluid/DXSystems/Cooling/AirSource/Data/Generic/Coil.mo

```diff
@@ -1,25 +1,68 @@
-within Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic;
-record DXCoil
+within Buildings.Fluid.DXSystems.Cooling.AirSource.Data.Generic;
+record Coil
   "Performance record for a DX Cooling Coil with one or multiple stages"
   extends Modelica.Icons.Record;
-  parameter Integer nSta(min=1) "Number of stages"
-    annotation (Evaluate = true,
-                Dialog(enable = not sinStaOpe));
-  parameter Real minSpeRat( min=0,max=1)=0.2 "Minimum speed ratio"
-    annotation (Dialog(enable = not sinStaOpe));
+
   final parameter Boolean sinStaOpe = nSta == 1
     "The data record is used for single speed operation"
     annotation(HideResult=true);
 
+  parameter Integer nSta(
+    final min=1)
+    "Number of stages"
+    annotation (Dialog(enable = not sinStaOpe));
+
+  parameter Real minSpeRat(
+    final min=0,
+    final max=1)=0.2
+    "Minimum speed ratio"
+    annotation (Dialog(enable = not sinStaOpe));
+
   replaceable parameter
-    Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.BaseClasses.Stage
-    sta[nSta]
-    constrainedby
-    Buildings.Fluid.HeatExchangers.DXCoils.AirSource.Data.Generic.BaseClasses.Stage
+    Buildings.Fluid.DXSystems.Cooling.AirSource.Data.Generic.BaseClasses.Stage sta[nSta]
     "Data record for coil performance at each stage";
+
   parameter Modelica.Units.SI.MassFlowRate m_flow_small=0.0001*sta[nSta].nomVal.m_flow_nominal
     "Small mass flow rate for regularization near zero flow"
     annotation (Dialog(group="Minimum conditions"));
+
+  /////////////////////////////////////////////////////////////////////////////////////
+  // Data added for type compatibility, but not used for calculations in cooling coils
+  final parameter Buildings.Fluid.DXSystems.Heating.BaseClasses.Types.DefrostOperation
+    defOpe=Buildings.Fluid.DXSystems.Heating.BaseClasses.Types.DefrostOperation.resistive
+    "Defrost operation type"
+    annotation(HideResult=true);
+
+  final parameter Buildings.Fluid.DXSystems.Heating.BaseClasses.Types.DefrostTimeMethods
+    defTri=Buildings.Fluid.DXSystems.Heating.BaseClasses.Types.DefrostTimeMethods.timed
+    "Type of method used to calculate defrost time fraction"
+    annotation(HideResult=true);
+
+  final parameter Real tDefRun(
+    final unit="1",
+    displayUnit="1")=0
+    "Time period fraction for which defrost cycle is run"
+    annotation(HideResult=true);
```

Comment:
time period fraction is not clear. Explain what it is a fraction off, what is the basis, e.g., what would a value of 1 mean?